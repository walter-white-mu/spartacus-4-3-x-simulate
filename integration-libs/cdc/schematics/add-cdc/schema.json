{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDCSchematics",
  "title": "CDC Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDC feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["CDC"]
    },
    "javascriptUrl": {
      "type": "string",
      "description": "Location of the CDC Java Script SDK in Customer Data Cloud",
      "x-prompt": "[CDC] What is the CDC JS SDK URL?"
    },
    "baseSite": {
      "type": "string",
      "description": "Name of the Spartcus base site",
      "x-prompt": "[CDC] What is the base site?",
      "default": "electronics-spa"
    },
    "sessionExpiration": {
      "type": "number",
      "description": "Session Expiration time in seconds",
      "x-prompt": "[CDC] What is the CDC Session expiration time (in seconds)?",
      "default": 3600
    }
  },
  "required": []
}
