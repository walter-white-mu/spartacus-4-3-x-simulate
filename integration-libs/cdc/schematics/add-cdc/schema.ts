import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  baseSite: string;
  javascriptUrl: string;
  sessionExpiration: number;
}
