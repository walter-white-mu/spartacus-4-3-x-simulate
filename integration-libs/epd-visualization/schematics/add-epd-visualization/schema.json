{
  "$schema": "http://json-schema.org/schema",
  "$id": "EpdVisualizationSchematics",
  "title": "EpdVisualization Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["EPD-Visualization"]
    },
    "baseUrl": {
      "type": "string",
      "description": "The base URL (origin) of the EPD Fiori Launchpad",
      "x-prompt": "[EPD Visualization] What is the base URL (origin) of your EPD Fiori Launchpad? e.g. https://mytenant.epd.cfapps.eu20.hana.ondemand.com"
    }
  },
  "required": []
}
