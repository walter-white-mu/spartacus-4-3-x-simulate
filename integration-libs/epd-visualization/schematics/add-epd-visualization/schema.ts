import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  baseUrl: string;
}
