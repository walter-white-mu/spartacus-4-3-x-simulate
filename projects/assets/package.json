{
  "name": "@spartacus/assets",
  "version": "4.3.7",
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/projects/assets",
  "license": "Apache-2.0",
  "scripts": {
    "build": "ng build assets --configuration production && yarn generate:translations:ts-2-json",
    "generate:translations:properties-2-ts": "ts-node ./generate-translations-properties-2-ts && cd ../.. && npx prettier \"./projects/assets/src/translations/**/*.ts\" --write",
    "generate:translations:ts-2-json": "ts-node ./generate-translations-ts-2-json",
    "generate:translations:ts-2-properties": "ts-node ./generate-translations-ts-2-properties"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
