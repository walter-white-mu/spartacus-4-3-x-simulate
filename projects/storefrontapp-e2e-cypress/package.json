{
  "name": "storefrontapp-e2e-cypress",
  "version": "3.0.0",
  "main": "index.js",
  "author": "SAP, Spartacus team",
  "license": "Apache-2.0",
  "scripts": {
    "cy:open": "cypress open",
    "cy:open:b2b": "cypress open --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers",
    "cy:open:epd-visualization": "cypress open --env BASE_SITE=powertools-epdvisualization-spa",
    "cy:open:ccv2": "cypress open --config baseUrl=https://spartacusstore.cg79x9wuu9-eccommerc1-p1-public.model-t.myhybris.cloud --env API_URL=https://api.cg79x9wuu9-eccommerc1-p1-public.model-t.myhybris.cloud",
    "cy:run": "cypress run",
    "cy:run:b2b": "cypress run --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers",
    "cy:run:product-configurator-vc": "cypress run --spec \"cypress/integration/**/*-configurator*.*e2e-spec.ts\"",
    "cy:run:ci": "cypress run --config-file cypress.ci.json --record --key $CYPRESS_KEY --tag \"b2c,all\" --parallel --group B2C --spec \"cypress/integration/!(vendor|b2b|ssr|cx_ccv2)/**/*.e2e-spec.ts\"",
    "cy:run:ci:b2b": "cypress run --config-file cypress.ci.json --env BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --record --key $CYPRESS_KEY --tag \"b2b,all\" --parallel --group B2B --spec \"cypress/integration/b2b/regression/**/*.e2e-spec.ts\"",
    "cy:run:cpq": "cypress run  --config-file cypress.ci.json --config baseUrl=https://jsapps.cpce-teamtiger1-d2-public.model-t.cc.commerce.ondemand.com,defaultCommandTimeout=100000,requestTimeout=1000000,responseTimeout=1000000,pageLoadTimeout=1000000 --reporter junit --reporter-options mochaFile=results/test-results-[hash].xml --env API_URL=https://api.cpce-teamtiger1-d2-public.model-t.cc.commerce.ondemand.com --spec \"cypress/integration/**/cpq-configuration.e2e-spec.ts\"",
    "cy:run:ci:ssr": "cypress run --config-file cypress.ci.json --config baseUrl=http://localhost:4000 --record --key $CYPRESS_KEY --tag \"ssr,all\" --parallel --group SSR --spec \"cypress/integration/ssr/pages.e2e-spec.ts\"",
    "cy:run:ci:ccv2": "cypress run --config-file cypress.ci.json --config baseUrl=$ENDPOINT_URL_PUBLIC_SPARTACUS --reporter junit --reporter-options mochaFile=results/spartacus-test-results-[hash].xml --env API_URL=$ENDPOINT_URL_PUBLIC_API --spec \"cypress/integration/cx_ccv2/regression/b2c/**/*.e2e-spec.ts\"",
    "cy:run:ci:ccv2-b2b": "cypress run --config-file cypress.ci.json --config baseUrl=$ENDPOINT_URL_PUBLIC_SPARTACUS --reporter junit --reporter-options mochaFile=results/spartacus-test-results-[hash].xml --env API_URL=$ENDPOINT_URL_PUBLIC_API,BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --spec \"cypress/integration/cx_ccv2/regression/b2b/**/*.e2e-spec.ts\"",
    "cy:run:ci:ccv2-cpq": "cypress run --config-file cypress.ci.json --config baseUrl=$ENDPOINT_URL_PUBLIC_SPARTACUS --reporter junit --reporter-options mochaFile=results/spartacus-test-results-[hash].xml --env API_URL=$ENDPOINT_URL_PUBLIC_API,BASE_SITE=powertools-spa,OCC_PREFIX_USER_ENDPOINT=orgUsers --spec \"cypress/integration/**/cpq-configuration.ccv2-e2e-spec.ts\"",
    "cy:run:ci:cds": "cypress run --config-file cypress.ci.json --record --key $CYPRESS_KEY --tag \"2011,b2c,all-cds\" --group CDS --spec \"cypress/integration/vendor/cds/**/*.e2e-spec.ts\"",
    "cy:run:ci:cdc": "cypress run --config-file cypress.ci.json --env API_URL=https://cdc-spa-2011-2102.demo.hybris.com --record --key $CYPRESS_KEY --tag \"2005,cdc\" --group CDC --spec \"cypress/integration/vendor/cdc/**/*.e2e-spec.ts\"",
    "cy:run:ci:digital-payments": "cypress run --config-file cypress.ci.json --env API_URL=https://backoffice.cp96avkh5f-sapcxteam1-d5-public.model-t.cc.commerce.ondemand.com --record --key $CYPRESS_KEY --tag \"2105,digital-payments\" --group DIGITAL-PAYMENTS --spec \"cypress/integration/vendor/digital-payments/*.e2e-spec.ts\"",
    "cy:run:ci:epd-visualization": "cypress run --config-file cypress.ci.json --env API_URL=https://api.cp96avkh5f-integrati1-d1-public.model-t.cc.commerce.ondemand.com,BASE_SITE=powertools-epdvisualization-spa --record --key $CYPRESS_KEY --tag \"2105,epd-visualization\" --group EPD_VISUALIZATION --ci-build-id $TRAVIS_BUILD_ID --spec \"cypress/integration/vendor/epd-visualization/**/*.e2e-spec.ts\""
  },
  "devDependencies": {
    "@cypress/webpack-preprocessor": "^5.9.1",
    "@testing-library/cypress": "^8.0.1",
    "cypress": "^8.5.0",
    "cypress-file-upload": "^5.0.8",
    "@babel/core": "^7.14.3",
    "@babel/preset-env": "^7.14.2",
    "babel-loader": "^8.2.2",
    "ts-loader": "^6.0.4",
    "typescript": "^4.2.4",
    "webpack": "^5.39.1"
  }
}
