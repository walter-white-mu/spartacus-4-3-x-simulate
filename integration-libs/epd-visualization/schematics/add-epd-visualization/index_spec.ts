/// <reference types="jest" />

import { RunSchematicTaskOptions } from '@angular-devkit/schematics/tasks/run-schematic/options';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CLI_EPD_VISUALIZATION_FEATURE,
  LibraryOptions,
  SpartacusOptions,
  SPARTACUS_SCHEMATICS,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';
import { Schema as SpartacusEpdVisualizationOptions } from './schema';

const collectionPath = path.join(__dirname, '../collection.json');
const featureModulePath =
  'src/app/spartacus/features/epd-visualization/epd-visualization-feature.module.ts';
const scssFilePath = 'src/styles/spartacus/epd-visualization.scss';

describe('Spartacus SAP EPD Visualization integration schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusEpdVisualizationOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
    baseUrl:
      'https://epd-acc-eu20-consumer.epdacc.cfapps.eu20.hana.ondemand.com',
  };

  const visualizationFeatureOptions: SpartacusEpdVisualizationOptions = {
    ...libraryNoFeaturesOptions,
    features: [CLI_EPD_VISUALIZATION_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        SPARTACUS_SCHEMATICS,
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', libraryNoFeaturesOptions, appTree)
        .toPromise();
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(featureModulePath)).toBeFalsy();
    });
  });

  describe('SAP EPD Visualization feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', visualizationFeatureOptions, appTree)
          .toPromise();
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });

      describe('styling', () => {
        it('should create a proper scss file', () => {
          const scssContent = appTree.readContent(scssFilePath);
          expect(scssContent).toMatchSnapshot();
        });

        it('should update angular.json', async () => {
          const content = appTree.readContent('/angular.json');
          expect(content).toMatchSnapshot();
        });
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should run the proper installation tasks', async () => {
        const tasks = schematicRunner.tasks
          .filter((task) => task.name === 'run-schematic')
          .map(
            (task) => task.options as RunSchematicTaskOptions<LibraryOptions>
          );
        expect(tasks.length).toEqual(0);
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...visualizationFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });

  describe('SAP EPD Visualization feature - No compilerOptions in tsconfig', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', visualizationFeatureOptions, appTree)
          .toPromise();
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should run the proper installation tasks', async () => {
        const tasks = schematicRunner.tasks
          .filter((task) => task.name === 'run-schematic')
          .map(
            (task) => task.options as RunSchematicTaskOptions<LibraryOptions>
          );
        expect(tasks.length).toEqual(0);
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...visualizationFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
