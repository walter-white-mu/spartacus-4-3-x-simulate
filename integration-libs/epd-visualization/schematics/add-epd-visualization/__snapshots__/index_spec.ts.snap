// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus SAP EPD Visualization integration schematics: ng-add SAP EPD Visualization feature - No compilerOptions in tsconfig eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { EpdVisualizationModule } from \\"@spartacus/epd-visualization\\";
import { epdVisualizationTranslationChunksConfig, epdVisualizationTranslations } from \\"@spartacus/epd-visualization/assets\\";
import { EpdVisualizationConfig, EpdVisualizationRootModule } from \\"@spartacus/epd-visualization/root\\";

@NgModule({
  declarations: [],
  imports: [
    EpdVisualizationRootModule,
    EpdVisualizationModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: epdVisualizationTranslations,
      chunks: epdVisualizationTranslationChunksConfig,
    },
  }),
  provideConfig(<EpdVisualizationConfig>{
    epdVisualization: {
      ui5: {
        bootstrapUrl: \\"https://sapui5.hana.ondemand.com/1.97.0/resources/sap-ui-core.js\\"
      },

      apis: {
        baseUrl: \\"https://epd-acc-eu20-consumer.epdacc.cfapps.eu20.hana.ondemand.com\\"
      }
    }
  })
  ]
})
export class EpdVisualizationFeatureModule { }
"
`;

exports[`Spartacus SAP EPD Visualization integration schematics: ng-add SAP EPD Visualization feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { EpdVisualizationModule } from \\"@spartacus/epd-visualization\\";
import { epdVisualizationTranslationChunksConfig, epdVisualizationTranslations } from \\"@spartacus/epd-visualization/assets\\";
import { EpdVisualizationConfig, EpdVisualizationRootModule } from \\"@spartacus/epd-visualization/root\\";

@NgModule({
  declarations: [],
  imports: [
    EpdVisualizationRootModule,
    EpdVisualizationModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: epdVisualizationTranslations,
      chunks: epdVisualizationTranslationChunksConfig,
    },
  }),
  provideConfig(<EpdVisualizationConfig>{
    epdVisualization: {
      ui5: {
        bootstrapUrl: \\"https://sapui5.hana.ondemand.com/1.97.0/resources/sap-ui-core.js\\"
      },

      apis: {
        baseUrl: \\"https://epd-acc-eu20-consumer.epdacc.cfapps.eu20.hana.ondemand.com\\"
      }
    }
  })
  ]
})
export class EpdVisualizationFeatureModule { }
"
`;

exports[`Spartacus SAP EPD Visualization integration schematics: ng-add SAP EPD Visualization feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { epdVisualizationTranslationChunksConfig, epdVisualizationTranslations } from \\"@spartacus/epd-visualization/assets\\";
import { EpdVisualizationConfig, EpdVisualizationRootModule, EPD_VISUALIZATION_FEATURE } from \\"@spartacus/epd-visualization/root\\";

@NgModule({
  declarations: [],
  imports: [
    EpdVisualizationRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [EPD_VISUALIZATION_FEATURE]: {
        module: () =>
          import('@spartacus/epd-visualization').then((m) => m.EpdVisualizationModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: epdVisualizationTranslations,
      chunks: epdVisualizationTranslationChunksConfig,
    },
  }),
  provideConfig(<EpdVisualizationConfig>{
    epdVisualization: {
      ui5: {
        bootstrapUrl: \\"https://sapui5.hana.ondemand.com/1.97.0/resources/sap-ui-core.js\\"
      },

      apis: {
        baseUrl: \\"https://epd-acc-eu20-consumer.epdacc.cfapps.eu20.hana.ondemand.com\\"
      }
    }
  })
  ]
})
export class EpdVisualizationFeatureModule { }
"
`;

exports[`Spartacus SAP EPD Visualization integration schematics: ng-add SAP EPD Visualization feature general setup styling should create a proper scss file 1`] = `"@import \\"@spartacus/epd-visualization\\";"`;

exports[`Spartacus SAP EPD Visualization integration schematics: ng-add SAP EPD Visualization feature general setup styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        },
        \\"@schematics/angular:application\\": {
          \\"strict\\": true
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/epd-visualization.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"500kb\\",
                  \\"maximumError\\": \\"2.5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"2kb\\",
                  \\"maximumError\\": \\"4kb\\"
                }
              ],
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"outputHashing\\": \\"all\\"
            },
            \\"development\\": {
              \\"buildOptimizer\\": false,
              \\"optimization\\": false,
              \\"vendorChunk\\": true,
              \\"extractLicenses\\": false,
              \\"sourceMap\\": true,
              \\"namedChunks\\": true
            }
          },
          \\"defaultConfiguration\\": \\"production\\"
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            },
            \\"development\\": {
              \\"browserTarget\\": \\"schematics-test:build:development\\"
            }
          },
          \\"defaultConfiguration\\": \\"development\\"
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/epd-visualization.scss\\"
            ],
            \\"scripts\\": []
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;
